From: Thomas Hindoe Paaboel Andersen <phomes@gmail.com>
Date: Fri, 31 Jan 2014 07:07:20 +0100
Subject: analyze: fix plot issues when using gummiboot

It would crash and the legend in the bottom followed the time 0.0.

(cherry picked from commit b5cfa7408c4cb68e9bb232fc34b07fd03c915617)
---
 src/analyze/systemd-analyze.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/analyze/systemd-analyze.c b/src/analyze/systemd-analyze.c
index f4c7f68..317baf9 100644
--- a/src/analyze/systemd-analyze.c
+++ b/src/analyze/systemd-analyze.c
@@ -540,7 +540,7 @@ static int analyze_plot(DBusConnection *bus) {
             name.nodename, name.release, name.version, name.machine);
 
         svg("<g transform=\"translate(%.3f,100)\">\n", 20.0 + (SCALE_X * boot->firmware_time));
-        svg_graph_box(m, -boot->firmware_time, boot->finish_time);
+        svg_graph_box(m, -(double) boot->firmware_time, boot->finish_time);
 
         if (boot->firmware_time) {
                 svg_bar("firmware", -(double) boot->firmware_time, -(double) boot->loader_time, y);
@@ -589,7 +589,10 @@ static int analyze_plot(DBusConnection *bus) {
                 y++;
         }
 
+        svg("</g>\n");
+
         /* Legend */
+        svg("<g transform=\"translate(20,100)\">\n");
         y++;
         svg_bar("activating", 0, 300000, y);
         svg_text(true, 400000, y, "Activating");
